src/targeting/session.ts

```diff
@@ -0,0 +1,173 @@
+import type { Participations } from '@guardian/ab-core';
+import type { CountryCode } from '@guardian/libs';
+import { isString } from '@guardian/libs';
+import type { False, True } from '../types';
+
+/* -- Types -- */
+
+const referrers = ['facebook', 'twitter', 'reddit', 'google'] as const;
+
+/**
+ * Session Targeting is based on the browser session
+ *
+ * Includes information such as the country of origin, referrer, page view ID.
+ *
+ * These values identify a browser session are either generated client-side,
+ * read from a cookie or passed down from the server.
+ */
+export type SessionTargeting = {
+	/**
+	 * **AB** Tests – [see on Ad Manager][gam]
+	 *
+	 * Type: _Dynamic_
+	 *
+	 * Values: typically start with `ab`
+	 *
+	 * [gam]: https://admanager.google.com/59666047#inventory/custom_targeting/detail/custom_key_id=186327
+	 */
+	ab: string[] | null;
```

Comment:
Is the `| null` important here? Or could we change this and the return type of `experimentsTargeting` to `string[]`?

Reply:
I think it’s nicer to have `null` than an empty array `[]`.
This means we can filter out nulls more easily in the future.